typed-racket-lib/typed-racket/types/generalize.rkt

```diff
@@ -28,6 +28,10 @@
         [(? (lambda (t) (subtype t -FloatComplex))) -FloatComplex]
         [(? (lambda (t) (subtype t -SingleFlonumComplex))) -SingleFlonumComplex]
         [(? (lambda (t) (subtype t -Number))) -Number]
+        [(? (lambda (t) (subtype t -Base-Regexp))) -Regexp]
```

Comment:
I think either this should generalize to `-Base-Regexp`, or they should both generalize to `-Regexp`.